Automattic/wp-calypso
# Environment Badge: add Store Sandbox dev helper

This adds a Store Sandbox dev helper to the environment badge in development and wpcalypso environments. It shows the current Store Sandbox status and allows the user to toggle Store Sandbox on/off.

<img width="258" alt="Screenshot 2024-06-10 at 11 04 36 PM" src="https://github.com/Automattic/wp-calypso/assets/942359/cf429084-7a3a-4b60-b3cf-8e6c75f47e2c">

Requires: D151699-code
See: https://github.com/Automattic/payments-shilling/issues/2466
Fixes: https://github.com/Automattic/payments-shilling/issues/2863

**To test:**
- with D151699-code applied, sandbox `public-api.wordpress.com`
- visit calypso.localhost:3000
- hover over the environment badge
- verify that you see the Store Sandbox helper
- if Store Sandbox is enabled (via the constant, cookie, or guest ticket method), verify that the Store Sandbox label is red
- if Store Sandbox is disabled, verify that the label matches the other dev helpers
- hover over the dev helper
- use the toggle to turn on/off Store Sandbox
